src/execution/mod.rs

```diff
@@ -411,26 +425,31 @@ fn run_until_done<A: Allocate, T: Timestamp>(
     worker: &mut Worker<A>,
     interrupt_flag: &AtomicBool,
     probe: ProbeHandle<T>,
-) {
+) -> PyResult<()> {
     let mut span = PeriodicSpan::new(Duration::from_secs(10));
     while !interrupt_flag.load(Ordering::Relaxed) && !probe.done() {
         worker.step();
         span.update();
+        Python::with_gil(|py| Python::check_signals(py)).map_err(|err| {
+            interrupt_flag.store(true, Ordering::Relaxed);
+            err
+        })?;
```

Comment:
Is this the piece that makes the error on `ctrl+c` finally readable? Good catch :+1:

Reply:
My issue here was that the `run_main` ctrl-c test was failing and I wasn't sure why. Especially because I could ctrl-c a long running example dataflow manually. I think this fixed it? I think there's something deeper we don't understand about signal management here. But if it works for now...